pkg/os/filesystem/types.go

```diff
@@ -0,0 +1,143 @@
+	// except for restrictions noted below. For details, please check:
+	// https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+	//
+	// Restrictions:
+	// Only absolute path (indicated by a drive letter prefix: e.g. "C:\") is accepted.
+	// Depending on the context parameter of this function, the path prefix needs
+	// to match the paths specified either as kubelet-csi-plugins-path
+	// or as kubelet-pod-path parameters of csi-proxy.
+	// UNC paths of the form "\\server\share\path\file" are not allowed.
+	// All directory separators need to be backslash character: "\".
+	// Characters: .. / : | ? * in the path are not allowed.
+	// Path cannot be a file of type symlink.
+	// Maximum path length will be capped to 260 characters.
+	Path string `protobuf:"bytes,1,opt,name=path,proto3" json:"path,omitempty"`
+	// Force remove all contents under path (if any).
+	Force bool `protobuf:"varint,2,opt,name=force,proto3" json:"force,omitempty"`
+}
+
+type LinkPathRequest struct {
+	// The path where the symlink is created in the host's filesystem.
+	// All special characters allowed by Windows in path names will be allowed
+	// except for restrictions noted below. For details, please check:
+	// https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+	//
+	// Restrictions:
+	// Only absolute path (indicated by a drive letter prefix: e.g. "C:\") is accepted.
+	// The path prefix needs needs to match the paths specified as
+	// kubelet-csi-plugins-path parameter of csi-proxy.
+	// UNC paths of the form "\\server\share\path\file" are not allowed.
+	// All directory separators need to be backslash character: "\".
+	// Characters: .. / : | ? * in the path are not allowed.
+	// source_path cannot already exist in the host filesystem.
+	// Maximum path length will be capped to 260 characters.
+	SourcePath string `protobuf:"bytes,1,opt,name=source_path,json=sourcePath,proto3" json:"source_path,omitempty"`
+	// Target path in the host's filesystem used for the symlink creation.
+	// All special characters allowed by Windows in path names will be allowed
+	// except for restrictions noted below. For details, please check:
+	// https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+	//
+	// Restrictions:
+	// Only absolute path (indicated by a drive letter prefix: e.g. "C:\") is accepted.
+	// The path prefix needs to match the paths specified as
+	// kubelet-pod-path parameter of csi-proxy.
+	// UNC paths of the form "\\server\share\path\file" are not allowed.
+	// All directory separators need to be backslash character: "\".
+	// Characters: .. / : | ? * in the path are not allowed.
+	// target_path needs to exist as a directory in the host that is empty.
+	// target_path cannot be a symbolic link.
+	// Maximum path length will be capped to 260 characters.
+	TargetPath string `protobuf:"bytes,2,opt,name=target_path,json=targetPath,proto3" json:"target_path,omitempty"`
+}
+
+type CreateSymlinkRequest struct {
```

Comment:
remove XXXRequest struct, and invoke directly in the safe_mounter_host_process.go would be simpler